src/ast.rs

```diff
@@ -7357,6 +7361,27 @@ impl Parse for SpecialName {
                 };
                 Ok((SpecialName::GuardTemporary(name, idx), tail))
             }
+            b"Gr" => {
+                let (resource_name_len, tail) = parse_number(10, false, tail)?;
+                debug_assert!(resource_name_len >= 0);
```

Comment:
Invalid input could give you a negative number here, no?

Reply:
I tried breaking it, and couldn't :(. Can you think of any invalid input that would break it?

Reply:
Ah you are passing false for `allow_signed` so I suppose I'm wrong.

(Also means you can leave the assert off, I think).

Reply:
Yup, that's redundant let me remove that.